include/nbl/asset/IAssetManager.h

```diff
@@ -342,19 +394,37 @@ class IAssetManager : public core::IReferenceCounted, public core::QuitSignallin
         {
             return getAssetInHierarchy(_file, _supposedFilename, _params,  0u, _override);
         }
-
         //TODO change name
         SAssetBundle getAsset(const std::string& _filename, const IAssetLoader::SAssetLoadParams& _params)
         {
             return getAsset(_filename, _params, &m_defaultLoaderOverride);
         }
-
         //TODO change name
         SAssetBundle getAsset(io::IReadFile* _file, const std::string& _supposedFilename, const IAssetLoader::SAssetLoadParams& _params)
         {
             return getAsset(_file, _supposedFilename, _params, &m_defaultLoaderOverride);
         }
 
+        SAssetBundle getAssetWholeBundleRestore(const std::string& _filename, const IAssetLoader::SAssetLoadParams& _params, IAssetLoader::IAssetLoaderOverride* _override)
+        {
+            return getAssetInHierarchyWholeBundleRestore(_filename, _params, 0u, _override);
+        }
+        //TODO change name
+        SAssetBundle getAssetWholeBundleRestore(io::IReadFile* _file, const std::string& _supposedFilename, const IAssetLoader::SAssetLoadParams& _params, IAssetLoader::IAssetLoaderOverride* _override)
+        {
+            return getAssetInHierarchyWholeBundleRestore(_file, _supposedFilename, _params, 0u, _override);
+        }
+        //TODO change name
+        SAssetBundle getAssetWholeBundleRestore(const std::string& _filename, const IAssetLoader::SAssetLoadParams& _params)
+        {
+            return getAssetWholeBundleRestore(_filename, _params, &m_defaultLoaderOverride);
+        }
+        //TODO change name
+        SAssetBundle getAssetWholeBundleRestore(io::IReadFile* _file, const std::string& _supposedFilename, const IAssetLoader::SAssetLoadParams& _params)
+        {
+            return getAssetWholeBundleRestore(_file, _supposedFilename, _params, &m_defaultLoaderOverride);
+        }
```

Comment:
for the public API user code.. this should be default behaviour of `getAsset`, no new functions.

Reply:
The philosophies...

User : Play it safe, Restore whole bundles, When inserting into cache insert with IMMUTABLE by default
Loader : Play it fast, Restore on a need-to-know basis, When inserting into cache insert with MUTABLE by default

Reply:
> When inserting into cache insert with IMMUTABLE by default

i thought we agreed on CPU_PERSISTENT

Reply:
yeah sorry I mis-wrote, yes user is CPU_PERSISTENT.